TeamNewPipe/NewPipe
# Show an alert-dialog when no appropriate file-manager was found

#### What is it?

#### Description of the changes in your PR
- See the linked issue for more details about the problem
- If no file-manager (or SAF compatible) file manager is installed on a device now an alert shows up

#### Before/After Screenshots/Screen Record
- Before: ``<App crashed>``
- After: 
![grafik](https://user-images.githubusercontent.com/40789489/143686395-93f0a7e6-d758-44b3-a97c-cbef08e1ad76.png)
![grafik](https://user-images.githubusercontent.com/40789489/145270635-c060610f-6dc7-4995-8095-1d38ce6e7e28.png) ← Android 10+


#### Fixes the following issue(s)
- Fixes #7113 and more than 10 other issues where users managed to have no file-manager.


#### APK testing 
The APK can be found by going to the "Checks" tab below the title. On the left pane, click on "CI", scroll down to "artifacts" and click "app" to download the zip file which contains the debug APK of this PR.

#### Due diligence

### How to test this?
To test this you have to get rid of all your file-managers (may led to data loss or other side effects). This may require disabling system apps.

I strongly recommend doing this on an emulator only!

How to do it on an emulated Android 10 device:
* Wipe all data from that device
* Open a shell/commandline
* Connect to the emulator using ``adb shell`` (ensure that you are connected to the correct device!)
* Get into the sudo mode: ``su``
* Disable Androids default file-manager: ``pm disable com.android.documentsui``
→ Now you can deploy NewPipe to the emulator and check if the changes work as expected
